Split contracts

biern opened the pull request:
- [x] Factor out common stuff to Lib.Eth
- [x] Split contracts
- [x] Standarize Contracts.SignHash naming
- [x] Get rid of ramda
- [x] Display contracts addresses as links
- [x] Import contracts from signhash-contracts instead of requiring json

Closes #58 
Closes #55

jksf commented:
e2e tests fail for one case:
```
 ✖ Signer with valid github source

   1) AssertionError: expected 'GitHub verification failed' to include 'foobar32167'

      Browser: Chrome 63.0.3239 / Linux 0.0.0

         31 |  const file = fixtures.signerFileWithValidGithub;
         32 |
         33 |  await t
         34 |    .setFilesToUpload('#file-upload', file.path)
         35 |    .expect(githubProof().textContent)
       > 36 |    .contains('foobar32167');
         37 |})
         38 |
         39 |
         40 |test('Signer with illegal github username', async t => {
         41 |  const file = fixtures.signerFileWithInvalidGithubProofValue;

         at Object.<anonymous> (/home/js/Projects/js/signhash/signhash-webapp/tests/e2e/tests/proofs.ts:36:6)
         at <anonymous> (/home/js/Projects/js/signhash/signhash-webapp/tests/e2e/tests/proofs.ts:7:71)
         at __awaiter (/home/js/Projects/js/signhash/signhash-webapp/tests/e2e/tests/proofs.ts:3:12)
         at test (/home/js/Projects/js/signhash/signhash-webapp/tests/e2e/tests/proofs.ts:30:51)
```
I updated `package-lock` by simply invoking `npm install`. I doubt I broke it :-)

biern commented:
You didn't break it :) Tests are failing now because in the coming PR I have updated the mnemonic so that it can be loaded in metamask, it wasn't long enough before. 
In consequence all the test addresses have changed. I've updated the test github account to use the new address and that's why the test in this PR is failing now. 
It should be fine to merge if you're concerned only about this test.
